### PR TITLE
EVG-7010 correctly retry docker setup job

### PR DESCRIPTION
**Pull request by john-m-liu**

### CODE REVIEW

File: units/provisioning_create_host.go [hunk 1]
```diff
@@ -353,11 +353,25 @@ func (j *createHostJob) isImageBuilt(ctx context.Context) (bool, error) {
 		return false, errors.Errorf("parent for host '%s' not running", j.host.Id)
 	}
 	if ok := parent.ContainerImages[j.host.DockerOptions.Image]; ok {
+		grip.Info(message.Fields{
```
Comment (tychoish) [inline units/provisioning_create_host.go]: do we really need this logging?

File: units/provisioning_create_host.go [hunk 2]
```diff
@@ -353,11 +353,25 @@ func (j *createHostJob) isImageBuilt(ctx context.Context) (bool, error) {
...
+			"image":    j.host.DockerOptions.Image,
+			"attempts": j.CurrentAttempt,
+			"job":      j.ID(),
+		})
 		return true, nil
 	}
 
 	//  If the image is not already present on the parent, run job to build the new image
 	if j.BuildImageStarted == false {
+		grip.Info(message.Fields{
```
Comment (tychoish) [inline units/provisioning_create_host.go]: should we further annotate the other logging statement in this if statement?

File: units/provisioning_create_host.go (hunk 1 above)
Comment (john-m-liu) [reply units/provisioning_create_host.go]: I find it useful. It should only log once per docker image, so I'd like to keep it in

File: units/provisioning_create_host.go (hunk 2 above)
Comment (john-m-liu) [reply units/provisioning_create_host.go]: It's not really a useful log message, but it is the only way that error is being handled so it's probably fine to keep as is

Comment (tychoish) [reply units/provisioning_create_host.go]: All errors that get j.AddError() are logged when the job finished, which I think should reduce the need to log errors in jobs.

Comment (john-m-liu) [reply units/provisioning_create_host.go]: That specific error is not added to the job because we don't want to retry if it happens, so we just log it

**Pull request merged**